backend/dataall/modules/redshift_datasets/services/redshift_connection_service.py

```diff
@@ -161,3 +174,30 @@ def _check_redshift_connection(account_id: str, region: str, connection: Redshif
                 f'Redshift database {connection.database} does not exist or cannot be accessed with these parameters: {e}'
             )
         return
+
+    @staticmethod
+    def _get_redshift_encryption(
+        account_id: str, region: str, connection: RedshiftConnection
+    ) -> RedshiftEncryptionType:
+        if connection.redshiftType == RedshiftType.Serverless.value:
+            namespace = redshift_serverless_client(account_id=account_id, region=region).get_namespace_by_id(
+                connection.nameSpaceId
+            )
+            return (
+                RedshiftEncryptionType.AWS_OWNED_KMS_KEY
+                if namespace.get('kmsKeyId', None) == RedshiftEncryptionType.AWS_OWNED_KMS_KEY.value
```

Comment:
do we not need to `describe_kms_key` similar to how we do with Provisioned Clusters L194 below?